recipes/x86_64_level/gen.py

```diff
@@ -0,0 +1,29 @@
+import archspec.cpu
+
+archs = archspec.cpu.TARGETS
+
+x86_64_levels = {}
+
+name_mapping = {
+  "x86_64": 1,
+  "x86_64_v2": 2,
+  "x86_64_v3": 3,
+  "x86_64_v4": 4,
+}
+
+for arch_name, arch in archs.items():
+    if arch.family.name != "x86_64":
+        continue
+    if arch.parents:
+        level = max(name_mapping.get(parent.name, 0) for parent in arch.ancestors)
+    else:
+        level = 1
+    x86_64_levels[arch_name] = level
```

Comment:
```
microarchitecture:
- x86_64
- x86_64_v2
- x86_64_v3
- x86_64_v4
...
level:
- 1
- 1
- 2
- 3
```
seemed strange. This should fix it:
```suggestion
    x86_64_levels[arch_name] = max(
        1,
        name_mapping.get(arch_name, 0),
        *(name_mapping.get(parent.name, 0) for parent in (arch.ancestors or ()))
    )
```